app/assets/javascripts/admin/addon/components/admin-page-header.gjs

```diff
@@ -41,14 +51,75 @@ export default class AdminPageHeader extends Component {
           <h1 class="admin-page-header__title">{{this.title}}</h1>
         {{/if}}
 
-        <div class="admin-page-header__actions">
-          {{yield
-            (hash
-              Primary=PrimaryButton Default=DefaultButton Danger=DangerButton
-            )
-            to="actions"
-          }}
-        </div>
+        {{#if (has-block "actions")}}
+          <div class="admin-page-header__actions">
+            {{#if this.site.mobileView}}
+              <DMenu
+                @identifier="admin-page-header-mobile-actions"
+                @title={{i18n "more_options"}}
+                @icon="ellipsis-vertical"
+                class="btn-small"
+              >
+                <:content>
+                  <DropdownMenu class="admin-page-header__mobile-actions">
+                    {{yield
+                      (hash
+                        Primary=PrimaryActionListItem
+                        Default=DefaultActionListItem
+                        Danger=DangerActionListItem
+                        Wrapped=WrappedActionListItem
+                      )
+                      to="actions"
+                    }}
+                  </DropdownMenu>
+                </:content>
+              </DMenu>
+            {{else}}
+              {{yield
+                (hash
+                  Primary=PrimaryButton
+                  Default=DefaultButton
+                  Danger=DangerButton
+                  Wrapped=WrappedButton
+                )
+                to="actions"
+              }}
+            {{/if}}
+          </div>
+        {{else if @headerActionComponent}}
+          <div class="admin-page-header__actions">
+            {{#if this.site.mobileView}}
+              <DMenu
+                @identifier="admin-page-header-mobile-actions"
+                @title={{i18n "more_options"}}
+                @icon="ellipsis-vertical"
+                class="btn-small"
+              >
+                <:content>
+                  <DropdownMenu class="admin-page-header__mobile-actions">
+                    <@headerActionComponent
+                      @actions={{hash
+                        Primary=PrimaryActionListItem
+                        Default=DefaultActionListItem
+                        Danger=DangerActionListItem
+                        Wrapped=WrappedActionListItem
+                      }}
+                    />
+                  </DropdownMenu>
+                </:content>
+              </DMenu>
+            {{else}}
+              <@headerActionComponent
+                @actions={{hash
+                  Primary=PrimaryButton
+                  Default=DefaultButton
+                  Danger=DangerButton
+                  Wrapped=WrappedButton
+                }}
+              />
+            {{/if}}
+          </div>
+        {{/if}}
```

Comment:
here's what I was thinking (it's 58 lines vs 69 before)

```suggestion
      {{#if (or (has-block "actions") @headerActionComponent)}}
        <div class="admin-page-header__actions">
          {{#if this.site.mobileView}}
            <DMenu
              @identifier="admin-page-header-mobile-actions"
              @title={{i18n "more_options"}}
              @icon="ellipsis-vertical"
              class="btn-small"
            >
              <:content>
                <DropdownMenu class="admin-page-header__mobile-actions">
                  {{#if (has-block "actions")}}
                    {{yield
                      (hash
                        Primary=PrimaryActionListItem
                        Default=DefaultActionListItem
                        Danger=DangerActionListItem
                        Wrapped=WrappedActionListItem
                      )
                      to="actions"
                    }}
                  {{else}}
                    <@headerActionComponent
                      @actions={{hash
                        Primary=PrimaryActionListItem
                        Default=DefaultActionListItem
                        Danger=DangerActionListItem
                        Wrapped=WrappedActionListItem
                      }}
                    />
                  {{/if}}
                </DropdownMenu>
              </:content>
            </DMenu>
          {{else}}
            {{#if (has-block "actions")}}
              {{yield
                (hash
                  Primary=PrimaryButton
                  Default=DefaultButton
                  Danger=DangerButton
                  Wrapped=WrappedButton
                )
                to="actions"
              }}
            {{else}}
              <@headerActionComponent
                @actions={{hash
                  Primary=PrimaryButton
                  Default=DefaultButton
                  Danger=DangerButton
                  Wrapped=WrappedButton
                }}
              />
            {{/if}}
          {{/if}}
        </div>
      {{/if}}
```

Reply:
Wait, I can golf it 😄 

48 lines!

```suggestion
      {{#if (or (has-block "actions") @headerActionComponent)}}
        <div class="admin-page-header__actions">
          {{#if this.site.mobileView}}
            <DMenu
              @identifier="admin-page-header-mobile-actions"
              @title={{i18n "more_options"}}
              @icon="ellipsis-vertical"
              class="btn-small"
            >
              <:content>
                <DropdownMenu class="admin-page-header__mobile-actions">
                  {{#let
                    (hash
                      Primary=PrimaryActionListItem
                      Default=DefaultActionListItem
                      Danger=DangerActionListItem
                      Wrapped=WrappedActionListItem
                    )
                    as |actions|
                  }}
                    {{#if (has-block "actions")}}
                      {{yield actions to="actions"}}
                    {{else}}
                      <@headerActionComponent @actions={{actions}} />
                    {{/if}}
                  {{/let}}
                </DropdownMenu>
              </:content>
            </DMenu>
          {{else}}
            {{#let
              (hash
                Primary=PrimaryButton
                Default=DefaultButton
                Danger=DangerButton
                Wrapped=WrappedButton
              )
              as |actions|
            }}
              {{#if (has-block "actions")}}
                {{yield actions to="actions"}}
              {{else}}
                <@headerActionComponent @actions={{actions}} />
              {{/if}}
            {{/let}}
          {{/if}}
        </div>
      {{/if}}
```

Reply:
Thank you for your service :saluting_face: :joy: